tools/helpers.ts

```diff
@@ -0,0 +1,239 @@
+import { prettier as prettierOpts } from '../package.json';
+
+const ruleRegex = /^S\d+/;
+export const DIRNAME = dirname(fileURLToPath(import.meta.url));
+export const TS_TEMPLATES_FOLDER = join(DIRNAME, 'templates', 'ts');
+export const JAVA_TEMPLATES_FOLDER = join(DIRNAME, 'templates', 'java');
+export const RULES_FOLDER = join(DIRNAME, '..', 'packages', 'jsts', 'src', 'rules');
+export const METADATA_FOLDER = join(
+  DIRNAME,
+  '..',
+  'sonar-plugin',
+  'javascript-checks',
+  'src',
+  'main',
+  'resources',
+  'org',
+  'sonar',
+  'l10n',
+  'javascript',
+  'rules',
+  'javascript',
+);
+export const header = await readFile(join(DIRNAME, 'header.ts'), 'utf8');
+
+const typeMatrix = {
+  CODE_SMELL: 'suggestion',
+  BUG: 'problem',
+  SECURITY_HOTSPOT: 'problem',
+  VULNERABILITY: 'problem',
+} as const;
+
+type rspecMeta = {
+  type: keyof typeof typeMatrix;
+  status: 'ready' | 'beta' | 'closed' | 'deprecated' | 'superseded';
+  title: string;
+  quickfix: 'covered' | undefined;
+  tags: string[];
+};
+
+const sonarWayProfile = JSON.parse(
+  await readFile(join(METADATA_FOLDER, `Sonar_way_profile.json`), 'utf-8'),
+);
+
+// Array sorter for Sonar rule IDs
+const getInt = (sonarKey: string) => parseInt(/^S(\d+)/.exec(sonarKey)[1]);
+const sonarKeySorter = (a: string, b: string) => getInt(a) - getInt(b);
+
+export function verifyRuleName(eslintId: string) {
+  const re = /^[a-z]+(-[a-z0-9]+)*$/;
+  if (!re.exec(eslintId)) {
+    throw new Error(`Invalid class name: it should match ${re}, but got "${eslintId}"`);
+  }
+}
+
+export function verifyRspecId(sonarKey: string) {
+  const re = /^S\d+$/;
+  if (!re.exec(sonarKey)) {
+    throw new Error(`Invalid rspec key: it should match ${re}, but got "${sonarKey}"`);
+  }
+}
+
+export function escapeRegExp(str: string) {
+  return str.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, '\\$1');
+}
+
+/**
+ * Inflate string template with given dictionary
+ * @param text template string
+ * @param dictionary object with the keys to replace
+ */
+export function inflateTemplate(text: string, dictionary: { [x: string]: string }): string {
+  for (const tok in dictionary) {
+    text = text.replace(new RegExp(escapeRegExp(tok), 'g'), dictionary[tok]);
+  }
+  return text;
+}
+
+/**
+ * Reads a template file, inflates it with the provided dictionary, and writes the prettified
+ * result to destination file
+ *
+ * @param templatePath path to template file
+ * @param dest destination file
+ * @param dict dictionary to inflate the template
+ */
+export async function inflateTemplateToFile(
+  templatePath: string,
+  dest: string,
+  dict: { [x: string]: string },
+) {
+  const template = await readFile(templatePath, 'utf8');
+  await writePrettyFile(dest, inflateTemplate(template, dict));
+}
+
+/**
+ * From the RSPEC json file, creates a generated-meta.ts file with ESLint formatted metadata
+ *
+ * @param sonarKey rule ID for which we need to create the generated-meta.ts file
```

Comment:
sonarKey is the SXXX format? Might be nice to add an example in the code
```suggestion
 * @param sonarKey rule ID ("SXXX") for which we need to create the generated-meta.ts file
```